File: Core/Java/Graph.java
class GraphNode<T> {
  public T value;
  public Object[] adjacent = new Object[1];

  GraphNode(T value) {
    this.value = value;
  }

  void addAdjacent(GraphNode<T> node) {
    this.adjacent = ArrayUtils.Append(node, this.adjacent);
  }
}

class BinaryNode {
  public int value;
  public BinaryNode left;
  public BinaryNode right;

  BinaryNode(int value) {
    this.value = value;
  }

  public String toString() {
    return this.value + " (l: " + this.leftValue() + ", r: " + this.rightValue() + ")";
  }

  public String leftValue() {
    if (this.left != null) {
      return "" + this.left.value;
    }
    return "";
  }

  public String rightValue() {
    if (this.right != null) {
      return "" + this.right.value;
    }
    return "";
  }
}

class Graph<T> {
  public T root;
}

File: Core/Swift/graph.swift
// Simple graph implementation.

// We must subclass "CustomStringConvertible" for our description var method to take effect.
class GraphNode: CustomStringConvertible {
  public var value: Any?;
  public var adjacent: [GraphNode] = [];
  public var marked = false;

  init(value: Any) {
    self.value = value;
  }

  public func addChild(node: GraphNode) -> Void {
    self.adjacent.append(node);
  }

  // Swift version of java toString override.
  public var description: String {
    var retStr = "GraphNode: \(self.value ?? "nil")"
    if (self.adjacent.count > 0) {
      retStr = "\(retStr)\n^adjacent:\n"
      for i in 0...(self.adjacent.count - 1) {
        retStr = "\(retStr)- GraphNode: \(self.adjacent[i].value ?? "nil")\n"
      }
    }
    return retStr;
  }
}

class Graph: CustomStringConvertible {
  public var nodes: [GraphNode] = [];

  public var description: String {
    var retStr = "";
    for node in self.nodes {
      retStr = "\(retStr)\(node.description)\n"
    }
    return retStr;
  }
}
